sdk/src/client/api/block_builder/input_selection/core/remainder.rs

```diff
@@ -22,8 +22,25 @@ use crate::{
 impl InputSelection {
     // Gets the remainder address from configuration of finds one from the inputs.
     fn get_remainder_address(&self) -> Option<(Address, Option<Bip44>)> {
-        if self.remainder_address.is_some() {
-            return self.remainder_address.map(|address| (address, None));
+        if let Some(remainder_address) = self.remainder_address {
+            // Search in inputs for the Bip44 chain for the remainder address, so the ledger can regenerate it
+            for input in self.available_inputs.iter().chain(self.selected_inputs.iter()) {
+                let alias_transition = is_alias_transition(
+                    &input.output,
+                    *input.output_id(),
+                    self.outputs.as_slice(),
+                    self.burn.as_ref(),
+                );
+                let required_address =
+                    input
+                        .output
+                        .required_and_unlocked_address(self.timestamp, input.output_id(), alias_transition);
```

Comment:
I know the current error can only come from the Treasury thing but I'm just thinking that if ever we add some more faillible stuff to be, we'd be silently ignoring it, maybe we should just make get_remainder_address faillible and propagate. It's all private, it's not the end of the world I guess

Reply:
Hm but even if there is another output in the future, we still wouldn't want to return an error, no?
I can also use `if let Ok(required_address)` if you prefer that

Reply:
I was not necessarily speaking about other kinds of outputs, just if for some reason we had something else that make sit faillible then we'd wouldn't notice here. I don't really see a downside making the whole thing fallible, so I'd say let's be on the safe side

Reply:
I'll approve anyway if you think it's not worth it

Reply:
Changed, but I think for other failure reasons we later might need extra handling then and ignore the error in this case